crates/testing/tests/consensus_task.rs

```diff
@@ -242,6 +242,353 @@ async fn test_consensus_vote_with_permuted_dac() {
+    let view = generator.current_view.unwrap();
+    // for view in (&mut generator).take(1) {
+    proposals.push(view.quorum_proposal.clone());
+    leaders.push(view.leader_public_key);
+    votes.push(view.create_quorum_vote(&handle));
+    // }
+
+    // This is a bog standard view and covers the situation where everything is going normally.
+    let view_1 = TestScriptStage {
+        inputs: vec![QuorumProposalRecv(proposals[0].clone(), leaders[0])],
+        outputs: vec![
+            exact(ViewChange(ViewNumber::new(1))),
+            exact(QuorumProposalValidated(proposals[0].data.clone())),
+            exact(QuorumVoteSend(votes[0].clone())),
+        ],
+        asserts: vec![is_at_view_number(1)],
+    };
+
+    // Fail twice here to "trigger" a view sync event. This is accomplished above by advancing the
+    // view number in the generator.
+    let view_2_3 = TestScriptStage {
+        inputs: vec![Timeout(ViewNumber::new(2)), Timeout(ViewNumber::new(3))],
+        outputs: vec![timeout_vote_send(), timeout_vote_send()],
+        // Times out, so we now have a delayed view
+        asserts: vec![is_at_view_number(1)],
+    };
+
+    // Handle the view sync finalize cert, get the requisite data, propose.
+    let cert = proposals[1].data.view_sync_certificate.clone().unwrap();
+
+    // Generate the timeout votes for the timeouts that just occurred.
+    let timeout_vote_view_2 = TimeoutVote::create_signed_vote(
+        TimeoutData {
+            view: ViewNumber::new(2),
+        },
+        ViewNumber::new(2),
+        &pub_key,
+        &priv_key,
+    )
+    .unwrap();
+
+    let timeout_vote_view_3 = TimeoutVote::create_signed_vote(
+        TimeoutData {
+            view: ViewNumber::new(3),
+        },
+        ViewNumber::new(3),
+        &pub_key,
+        &priv_key,
+    )
+    .unwrap();
+
+    let view_4 = TestScriptStage {
+        inputs: vec![
+            TimeoutVoteRecv(timeout_vote_view_2),
+            TimeoutVoteRecv(timeout_vote_view_3),
+            ViewSyncFinalizeCertificate2Recv(cert),
+            QuorumProposalRecv(proposals[1].clone(), leaders[1]),
+            SendPayloadCommitmentAndMetadata(payload_commitment, (), ViewNumber::new(4)),
+        ],
+        outputs: vec![
+            exact(ViewChange(ViewNumber::new(4))),
+            exact(QuorumProposalValidated(proposals[1].data.clone())),
+            quorum_proposal_send(),
+        ],
+        asserts: vec![is_at_view_number(4)],
+    };
+
+    let consensus_state = ConsensusTaskState::<
+        TestTypes,
+        MemoryImpl,
+        SystemContextHandle<TestTypes, MemoryImpl>,
+    >::create_from(&handle)
+    .await;
+
+    let stages = vec![view_1, view_2_3, view_4];
+
+    inject_consensus_polls(&consensus_state).await;
+    run_test_script(stages, consensus_state).await;
+}
+
+#[cfg(test)]
+#[cfg_attr(
+    async_executor_impl = "tokio",
+    tokio::test(flavor = "multi_thread", worker_threads = 2)
```

Comment:
nit, we can get rid of `worker_threads = 2` it just limits tokio executor to 2 threads arbitrarily

Reply:
Done!